opentracing-spring-web-autoconfigure/src/test/java/io/opentracing/contrib/spring/web/autoconfig/TracerAutoConfigurationTest.java

```diff
@@ -4,35 +4,64 @@
 import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertTrue;
 
+import java.lang.reflect.Field;
+
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 
+import io.opentracing.NoopTracerFactory;
 import io.opentracing.Tracer;
+import io.opentracing.mock.MockTracer;
 import io.opentracing.util.GlobalTracer;
+import io.opentracing.util.ThreadLocalActiveSpanSource;
 
 @SpringBootTest(
         classes = {TracerAutoConfigurationTest.SpringConfiguration.class})
 @RunWith(SpringJUnit4ClassRunner.class)
 public class TracerAutoConfigurationTest {
 
+    // Temporary solution until https://github.com/opentracing/opentracing-java/issues/170 resolved
+    private static void _setGlobal(Tracer tracer) {
+        try {
+            Field globalTracerField = GlobalTracer.class.getDeclaredField("tracer");
+            globalTracerField.setAccessible(true);
+            globalTracerField.set(null, tracer);
```

Comment:
it should set the tracer to noop not null

Reply:
It is being set to NoopTracer (see https://github.com/opentracing-contrib/java-spring-web/pull/44/files/882f6131b5bc4552ea2d54dbdcb8a6391a467158#diff-b993f7b05949a5bf1ea88ba95f2cd8f1R44) - the null is the target object, null in this case because the field is static.